### PR TITLE
Fix APIs Being Called Before Input Is Provided

### LINKED ISSUE #891: Fix APIs being called before input is provided
API calls for searching tasks and searching users are being made before input is provided

image
image

You will be able to reproduce this in local, with super user access

### PR DESCRIPTION
**Pull request by ajoykumardas12**
Closes #891 
### Issue:
- #891 
### Description:
On Issues page api call for issues and assignee are being done before any input is provided. During assignee suggestion for task assignment api calls are being made even when assignee input is empty.

### Feature Flag:
This is a fix change that affects both features under feature flag and out of feature flag.

### Frontend Changes
- [x] Yes
- [ ] No

### Backend Changes
- [ ] Yes
- [x] No

### Dev Tested:
- [x] Yes
- [ ] No

Screenshot_test

### Images/video of the change:
Before:
Screenshot_before

After:
Screenshot_after

https://github.com/Real-Dev-Squad/website-status/assets/87164140/2e2ba175-e54d-423f-a189-f0eb12113973

### CODE REVIEW

File: __tests__/Unit/pages/issues/Issues.test.tsx [hunk 1]
```diff
@@ -35,9 +35,7 @@ describe('Issues Page', () => {
         const buttonElement = getByRole('button', buttonAttribute);
         expect(buttonElement).toBeInTheDocument();
 
-        // button should be disabled when the data is loading
-        const loadingElement = getByText(loadingText);
-        expect(loadingElement).toBeInTheDocument();
```
Comment (Ajeyakrishna-k) [inline __tests__/Unit/pages/issues/Issues.test.tsx:40]: why was this removed?

Comment (ajoykumardas12) [reply __tests__/Unit/pages/issues/Issues.test.tsx:40]: Earlier `fetchIssues()` with empty search term was being called on mount whose loading state would render the loadingText.
This is removed now, so currently there is no loadingText on first load.

**Review by Ajeyakrishna-k — APPROVED** [review]
